mesop/utils/url_utils.py

```diff
@@ -18,4 +20,25 @@ def _escape_url_for_csp(url: str) -> str:
 
 
 def sanitize_url_for_csp(url: str) -> str:
-  return _escape_url_for_csp(_remove_url_query_param(url))
+  return _escape_url_for_csp(remove_url_query_param(url))
+
+
+def get_url_query_params(url: str) -> list[pb.QueryParam]:
+  query_param = urlparse(url).query
+  param_dict: dict[str, list[str]] = {}
+  if query_param:
+    for param in query_param.split("&"):
+      if "=" in param:
+        key, value = param.split("=", 1)
+        if key in param_dict:
+          param_dict[key].append(value)
+        else:
+          param_dict[key] = [value]
+      else:
+        # Handle cases where there's a key without a value
+        param_dict[param] = []
```

Comment:
For the case of `var1=&var2=2, I noticed that that var1 would become var1='' empty string `{'var1': ['']}`. Is that expected behavior -- I guess the differentiator between `var1&` and `var1=&`?

Reply:
I think so - I tested out the browser API and also get an empty string:
```js
new URLSearchParams('?a=&b=1').forEach((value, key) => {console.log('value', value, 'key', key)})
```